src/components/Ranking/Ranking.js

```diff
@@ -0,0 +1,80 @@
+import * as classes from './style.module.scss';
+
+export const RANKING_COMPONENT_NAME = 'ranking-component';
+
+class Ranking extends HTMLElement {
+  constructor() {
+    super();
+
+    this._contents = new DocumentFragment();
+
+    this.container = Ranking.addContainer();
+
+    this.titleElement = Ranking.addTitle();
+    this.container.appendChild(this.titleElement);
+
+    this.rankingList = Ranking.createRankingList(this.playerScores);
+    this.container.appendChild(this.rankingList);
+
+    this._contents.appendChild(this.container);
+  }
+
+  connectedCallback() {
+    this.appendChild(this._contents);
+  }
+
+  setUserScores(playerScores) {
+    this.playerScores = playerScores;
+  }
+
+  static addContainer() {
+    const container = document.createElement('div');
+    container.id = classes.RankingContainer;
+    return container;
+  }
+
+  static addTitle() {
+    const title = document.createElement('header');
+    title.id = classes.RankingTitle;
+    title.textContent = 'Mode Ranking';
+    return title;
+  }
+
+  static createRankingList(playerScores) {
+    const rankingList = document.createElement('div');
+
+    let playerRanking = [
+      { place: '1st' },
+      { place: '2nd' },
+      { place: '3rd' },
+    ].reduce((acc, item, index) => {
+      return [...acc, { ...item, ...playerScores[index] }];
```

Comment:
For better performance you can use acc.push({...item, ...playerScores[index]}] as it won't recreate acc object for each iteration

Reply:
After doing so (`acc.push({ ...item, ...playerScores[index] })`) I get an error: 
```
TypeError: can't access property "push", acc is undefined
```

Reply:
Ok, I guess you have returned the same line. Make it work like this:
```
{
    acc.push({ ...item, ...playerScores[index] });
    return acc
}
```

Reply:
Ok, it works now.